packages/node_modules/@ciscospark/spark-core/src/lib/credentials/credentials.js

```diff
@@ -404,15 +404,18 @@ const Credentials = SparkPlugin.extend({
         }
         this.userTokens.reset();
         this.supertoken = st;
-        return Promise.all(tokens.map((token) => {
-          this.logger.info(`credentials: revoking token for ${token.scope}`);
-          return this.downscope(token.scope)
-            // eslint-disable-next-line max-nested-callbacks
-            .then((t) => {
-              this.userTokens.add(t);
+        return Promise.all(tokens.map((token) => this.downscope(token.scope)
+          // eslint-disable-next-line max-nested-callbacks
+          .then((t) => {
+            this.userTokens.add(t);
+            this.logger.info(`credentials: revoking token for ${token.scope}`);
+            try {
```

Comment:
How were you seeing this fail? I would have expected a rejected promise, not a throw exception.

Reply:
i think it throws an exception here. I debugged it with the dev tools and it fails at this line in `token.revoke()`: https://github.com/ciscospark/spark-js-sdk/blob/master/packages/node_modules/@ciscospark/spark-core/src/lib/credentials/token.js#L418

<img width="457" alt="screen shot 2018-01-05 at 6 40 02 pm" src="https://user-images.githubusercontent.com/819823/34755211-6ee9b1ea-f577-11e7-9dae-b88ee05bbc6d.png">

Reply:
Gotcha. @tran2 does https://github.com/ciscospark/spark-js-sdk/pull/796 fix this? I feel like you were looking into this error before Christmas, but I can't remember exactly what that PR addresses.

Reply:
@ianwremmel PR #796 should fix the issue where token.revoke() fails due to logout not removing all the tokens

Reply:
👍 #796 fixes underlying issue because token is not cleaned up when logging out. Added this in case token.revoke() fails for different reasons, but i'm okay to close this PR if it's not necessary

Reply:
If somehow token was in a bad state, token.revoke() would fail. I think it makes sense to handle when it fails but I'm not sure how since I'm not familiar with this flow:
- Why was the revoke necessary here?
- If `token.revoke()` fails, the token won't get revoked (by sending a request to the backend) would there be any security issues since it might still be in the local storage?

Reply:
@ianwremmel? do you think this is still necessary? I think revoke was called here to revoke the previous token after getting the new token. maybe we can remove it from localStorage if it fails... i think it calls userTokens.reset prior to this, but @ianwremmel mentioned that that may not remove it from localStorage.